### ISSUE TITLE
Build error with x86_64 3.14.19

### ISSUE THREAD

**Issue by RayBloodworth** [sigboe/gslX68X#1]
When I build this driver with kernel version 3.14.19, some errors occured:
[  179s] drivers/input/touchscreen/silead.c: In function 'silead_ts_read_data':
[  179s] drivers/input/touchscreen/silead.c:210:3: error: too many arguments to function 'input_mt_assign_slots'
[  179s]    input_mt_assign_slots(data->input, data->slots, data->pos,
[  179s]    ^
[  179s] In file included from drivers/input/touchscreen/silead.c:27:0:
[  179s] include/linux/input/mt.h:120:5: note: declared here
[  179s]  int input_mt_assign_slots(struct input_dev _dev, int *slots,
[  179s]      ^
[  179s] drivers/input/touchscreen/silead.c: In function 'silead_ts_read_props':
[  179s] drivers/input/touchscreen/silead.c:432:2: error: implicit declaration of function 'device_property_read_u16' [-Werror=implicit-function-declaration]
[  179s]   error = device_property_read_u16(dev, SILEAD_DP_X_MAX, &data->x_max);
[  179s]   ^
[  179s] drivers/input/touchscreen/silead.c:444:2: error: implicit declaration of function 'device_property_read_u8' [-Werror=implicit-function-declaration]
[  179s]   error = device_property_read_u8(dev, SILEAD_DP_MAX_FINGERS,
[  179s]   ^
[  179s] drivers/input/touchscreen/silead.c:451:2: error: implicit declaration of function 'device_property_read_string' [-Werror=implicit-function-declaration]
[  179s]   error = device_property_read_string(dev, SILEAD_DP_FW_NAME,
[  179s]   ^
[  179s] drivers/input/touchscreen/silead.c:456:2: error: implicit declaration of function 'device_property_read_bool' [-Werror=implicit-function-declaration]
[  179s]   data->x_invert = device_property_read_bool(dev, SILEAD_DP_X_INVERT);
[  179s]   ^
[  179s] drivers/input/touchscreen/silead.c: In function 'silead_ts_probe':
[  179s] drivers/input/touchscreen/silead.c:534:7: error: 'GPIOD_OUT_LOW' undeclared (first use in this function)
[  179s]        GPIOD_OUT_LOW, 1);
[  179s]        ^
[  179s] drivers/input/touchscreen/silead.c:534:7: note: each undeclared identifier is reported only once for each function it appears in
[  179s] drivers/input/touchscreen/silead.c:533:21: error: too many arguments to function 'devm_gpiod_get_index'
[  179s]   data->gpio_power = devm_gpiod_get_index(dev, SILEAD_PWR_GPIO_NAME,
[  179s]                      ^
[  179s] In file included from drivers/input/touchscreen/silead.c:23:0:
[  179s] include/linux/gpio/consumer.h:31:32: note: declared here
[  179s]  struct gpio_desc *__must_check devm_gpiod_get_index(struct device *dev,
[  179s]                                 ^
[  179s] cc1: some warnings being treated as errors
[  179s] scripts/Makefile.build:308: recipe for target 'drivers/input/touchscreen/silead.o' failed
[  179s] make[3]: *_\* [drivers/input/touchscreen/silead.o] Error 1
[  179s] scripts/Makefile.build:455: recipe for target 'drivers/input/touchscreen' failed
[  179s] make[2]: **\* [drivers/input/touchscreen] Error 2
[  179s] scripts/Makefile.build:455: recipe for target 'drivers/input' failed
[  179s] make[1]: **\* [drivers/input] Error 2
[  179s] make[1]: **\* Waiting for unfinished jobs....

Is this kernel version to old?

**Comment by sigboe** [comment]
Yes I think the required version is somewhere in the 3.19 space. So I suggest 4.2 or newer. Do you have the ability to try a newer kernel?

**Comment by RayBloodworth** [comment]
No, 3.14 is the newest version I can use...
Is it possible to modify this driver to support 3.14 ?

**Comment by sigboe** [comment]
Oh, I get this "Is this possible" question a lot at work, yes its possible. But I believe this driver uses some functions that depend on a newer kernel version, replacement fictions would need to be rewritten. I don't have the skills for this, but just in case; what commands did you use to fetch and compile this driver?

**Comment by RayBloodworth** [comment]
I got kernel source code from Tizen and build it with gbs command. Firt I add this driver into kernel source code and add Kconfig & Makefile, then create the deconfig to add this driver into kernel . After this, I use make deconfig & make zImage ARCH=x86_64 to build the kernel.
